src/main/java/mate/academy/springboot/swagger/service/ProductServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.springboot.swagger.service;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    private final ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public Product create(Product product) {
+        return productRepository.save(product);
+    }
+
+    @Override
+    public Product getById(Long id) {
+        return productRepository.getById(id);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        productRepository.deleteById(id);
+    }
+
+    @Override
+    public void update(Product product) {
+        Product newProduct = new Product();
+        newProduct.setId(product.getId());
+        newProduct.setTitle(product.getTitle());
+        newProduct.setPrice(product.getPrice());
+        productRepository.save(newProduct);
+    }
```

Comment:
there is no need to create a new product
```suggestion
    @Override
    public void update(Product product) {
        productRepository.save(newProduct);
    }
```